components/partial/IndexPageGrid.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import styled from "@emotion/styled";
+import { useRouter } from "next/router";
+import Link from "next/link";
+import Banner from "../base/Banner";
+import FilterTabs from "../base/FilterTabs";
+import CardGrid from "../base/CardGrid";
+import Card from "../base/Card";
+import { slugify } from "../../modules/helpers";
+
+const BannerWrap = styled.div`
+  margin: -20px -40px 40px -40px;
+`;
+
+const CardGridWrap = styled.div`
+  margin-bottom: 20px;
+`;
+
+type CardType = {
+  topic: string;
+  description: string;
+  icon: string;
+  category: string;
+  href: string;
+};
+type Props = {
+  unfilteredCards: Array<CardType>;
+  bannerProps: {
+    icon: string;
+    topic: string;
+    description: string;
+  };
+};
+
+const IndexPageGrid = ({ unfilteredCards, bannerProps }: Props) => {
+  const router = useRouter();
+  const filters: any = [
+    { value: null, label: "All" },
+    ...new Set(unfilteredCards.map((card) => card.category)),
+  ];
+  const filteredCards = unfilteredCards.filter(
+    (card) =>
+      !router.query.filter || router.query.filter === slugify(card.category)
+  );
+
+  const setFilter = (selectedFilter) => {
+    if (selectedFilter.value) {
+      router.push({
+        pathname: router.pathname,
+        query: { filter: selectedFilter.value },
+      });
+    } else {
+      router.push(router.pathname);
+    }
+  };
```

Comment:
Using url query params (rather than state) to track the selected filter. This has the benefit of being able to link directly to filtered results